profit_analysis/analysis.py

```diff
@@ -0,0 +1,222 @@
+import datetime
+
+import pandas as pd
+import web3
+from profit_analysis.block_utils import add_block_timestamp
+from profit_analysis.coingecko import (
+    add_cg_ids,
+    get_address_to_coingecko_ids_mapping,
+    get_coingecko_historical_prices,
+)
+from profit_analysis.column_names import (
+    AMOUNT_DEBT_KEY,
+    AMOUNT_RECEIVED_KEY,
+    CG_ID_DEBT_KEY,
+    CG_ID_RECEIVED_KEY,
+    DECIMAL_DEBT_KEY,
+    PRICE_DEBT_KEY,
+    PRICE_KEY,
+    PRICE_RECEIVED_KEY,
+    TIMESTAMP_KEY,
+    TOKEN_DEBT_KEY,
+    TOKEN_RECEIVED_KEY,
+)
+from profit_analysis.constants import DATA_PATH
+from profit_analysis.read import read_profit_from_to
+from profit_analysis.token_utils import get_decimals
+
+"""
+Steps:
+1. given blockfrom and block to, read the profit
+"""
+
+WETH_TOKEN_ADDRESS = "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619"
+PD_DATETIME_FORMAT = "datetime64[ns]"
+
+
+def analyze_profit(inspect_db_session, block_from, block_to, save_to_csv=False):
+    profit = read_profit_from_to(inspect_db_session, block_from, block_to)
+    w3 = create_web3()
+    profit = add_block_timestamp(w3, profit)
+    profit = add_cg_ids(profit)
+    profit = get_usd_profit_arbitrages(profit, save_to_csv)
+    return profit
+
+
+def get_usd_profit_arbitrages(profit_by_block, save_to_csv=False):
```

Comment:
Is it possible to add a quick description of the method ? From a high-level perspective, no need to add lots of detail.
As in: "First, we ... Then, using the results from ... we ... and finally we can ..."
No need to explain edge cases (error management is explicit).

Reply:
:heavy_check_mark: